cli/npm/global.d.ts

```diff
@@ -0,0 +1,44 @@
+type Rule =
+  | string
+  | RegExp
+  | {
+      /** @internal */
+      _ruleBrand: never;
```

Comment:
What's does the internal `_ruleBrand` do?

Reply:
I  am not implementing details here, for  it seems to be unnecessary. So I make use of nominal typing to prevent mistakes. Typescript does not provide nominal typing and this is a trick to get something similar with nominal typing. You can refer https://spin.atomicobject.com/2018/01/15/typescript-flexible-nominal-typing/